tests/unit/OptionsReadRepositoryConfigFileTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Vipgoci\Tests\Options;
+
+use PHPUnit\Framework\TestCase;
+
+require_once( __DIR__ . '/mock/OptionsReadRepositoryConfigFile.php' );
+require_once( __DIR__ . '/../../misc.php' );
+require_once( __DIR__ . '/../../options.php' );
+
+// phpcs:disable PSR1.Files.SideEffects
+
+final class OptionsReadRepositoryConfigFileTest extends TestCase {
+
+	/**
+	 * @covers ::vipgoci_options_read_repo_file
+	 *
+	 * Tests against the --lint-only-modified-files option.
+	 */
+	public function testIfOptionLintOnlyModifiedFilesOptionIsReplacedTest(): void {
```

Comment:
I think this code is fine.

However, I'm wondering if this test is needed? The `vipgoci_options_read_repo_file()` function itself is tested in a [separate test](https://github.com/Automattic/vip-go-ci/blob/3e4b7e455d16a2d70306019facbad6285252b863/tests/OptionsReadRepoFileTest.php), and this includes testing boolean functions. Does the current test add any further testing of functionality? I'm just wondering if the test is re-testing something already being tested?

Reply:
This is only executing a unit test, using mocks, while the `vipgoci_options_read_repo_file` test executes an integration test, reading from the actual configuration file. I would keep both since this test can be run locally easily without external dependencies.